packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -491,16 +502,27 @@ def __init__(
         """
         AbciAppDB._check_data(setup_data)
         self._setup_data = deepcopy(setup_data)
-        self._cross_period_persisted_keys: Set[str] = (
-            cross_period_persisted_keys.copy()
-            if cross_period_persisted_keys is not None
-            else set()
-        )
         self._data: Dict[int, Dict[str, List[Any]]] = {
             RESET_COUNT_START: self.setup_data  # the key represents the reset index
         }
         self._round_count = ROUND_COUNT_DEFAULT  # ensures first round is indexed at 0!
 
+        self._cross_period_persisted_keys = self.default_cross_period_keys.union(
+            cross_period_persisted_keys or frozenset()
+        )
+        self._cross_period_check()
+
+    def _cross_period_check(self) -> None:
+        """Check the cross period keys against the setup data."""
+        not_in_cross_period = set(self._setup_data).difference(
+            self.cross_period_persisted_keys
+        )
+        if not_in_cross_period:
+            _logger.warning(
+                f"The setup data ({self._setup_data.keys()}) contain keys that are not in the "
+                f"cross period persisted keys ({self.cross_period_persisted_keys}): {not_in_cross_period}"
```

Comment:
Hm noisy?

Reply:
We currently have all the setup data in the default cross-period keys, so this should not warn. In the future, we might extend the `setup` data and forget to update the default cross-period keys to contain the new item(s). At this point, it would start warning and the app would also fail if someone tried to use the new item(s) in a period > 0. This warning might be useful to troubleshoot faster :thinking: 

What do you think?